code/modules/antagonists/cult/cult_items.dm

```diff
@@ -106,6 +108,129 @@ Striking a noncultist, however, will tear their flesh."}
+	// Lock
+	PATH_LOCK = list(WIELDER_SPELL = /datum/action/cooldown/spell/pointed/burglar_finesse, \
+	SWORD_SPELL = /datum/action/cooldown/spell/pointed/apetra_vulnera, SWORD_PREFIX = "incisive"), \
+	// Moon
+	PATH_MOON = list(WIELDER_SPELL = /datum/action/cooldown/spell/pointed/projectile/moon_parade, \
+	SWORD_SPELL = /datum/action/cooldown/spell/pointed/moon_smile, SWORD_PREFIX = "shimmering"), \
+	// Starter
+	PATH_START = list(WIELDER_SPELL = null, SWORD_SPELL = null, SWORD_PREFIX = "nascent") // lol loser
+))
+
+/obj/item/melee/cultblade/haunted
+	name = "haunted longsword"
+	desc = "An eerie sword with a blade that is less 'black' than it is 'absolute nothingness'. It glows with furious, restrained green energy."
+	icon_state = "hauntedblade"
+	inhand_icon_state = "hauntedblade"
+	worn_icon_state = "hauntedblade"
+	force = 35
+	throwforce = 15
+	block_chance = 55
+	wound_bonus = -25
+	bare_wound_bonus = 30
+	free_use = TRUE
+	light_color = COLOR_BLACK
+	light_system = OVERLAY_LIGHT
+	light_range = 4
+	// holder for the actual action when created.
+	var/datum/action/cooldown/spell/path_wielder_action
+	var/mob/living/trapped_entity
+	var/heretic_path
+
+/obj/item/melee/cultblade/haunted/Initialize(mapload, mob/soul_to_bind, mob/awakener, no_binding)
+	. = ..()
+
+	if(!no_binding || mapload)
+		bind_soul(soul_to_bind, awakener)
+	ADD_TRAIT(src, TRAIT_CASTABLE_LOC, INNATE_TRAIT)
+	AddElement(/datum/element/heretic_focus)
+
+/obj/item/melee/cultblade/haunted/proc/bind_soul(mob/soul_to_bind, mob/awakener)
+
+	var/datum/mind/trapped_mind = soul_to_bind.mind
+
+	if(soul_to_bind)
+		AddComponent(/datum/component/spirit_holding,\
+			soul_to_bind = soul_to_bind.mind,\
+			awakener = awakener,\
+			allow_renaming = FALSE,\
+		)
+
+	// Get the heretic's new body and antag datum.
+	trapped_entity = trapped_mind.current
+	var/datum/antagonist/heretic/heretic_holder = IS_HERETIC(trapped_entity)
+	if(!heretic_holder)
+		CRASH("[soul_to_bind] not a heretic on the heretic soul blade.")
+
+	// Set the sword's path for spell selection.
+	heretic_path = heretic_holder.heretic_path
+
+	// Copy the objectives to keep for roundend, remove the datum as neither us nor the heretic need it anymore
+	var/list/copied_objectives = heretic_holder.objectives.Copy()
+	trapped_entity.mind.remove_antag_datum(/datum/antagonist/heretic)
+
+	// Add the fallen antag datum, give them a heads-up of what's happening.
+	var/datum/antagonist/soultrapped_heretic/fallen = trapped_entity.mind.add_antag_datum(/datum/antagonist/soultrapped_heretic)
+	fallen.objectives = copied_objectives
+	to_chat(trapped_entity, span_alert("You've been sacrificed to the Enemy, and trapped inside a haunted blade! While you cannot escape, you may decide by yourself to be a nuisance with the few abilities you still have, or aid whoever wields it."))
+	// Unrelated bug: Objectives dont show up on the you are X thing as they are varsetted post-facto
```

Comment:
I wouldn't say it's a bug, but I think you can circumvent this by `new`ing the datum, adding the objectives, and then adding the datum with the instance you made